jetty-servlet/src/test/java/org/eclipse/jetty/servlet/InitServletTest.java

```diff
@@ -0,0 +1,152 @@
+import org.eclipse.jetty.server.Server;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.array;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class InitServletTest
+{
+
+    public static class DemoServlet extends HttpServlet
+    {
+        AtomicInteger initCount = new AtomicInteger();
+
+        @Override
+        public void init() throws ServletException
+        {
+            super.init();
+            try
+            {
+                //Make the initialization last a little while so
+                //other request can run
+                Thread.sleep(5000);
+            }
+            catch (InterruptedException e)
+            {
+                throw new ServletException(e);
+            }
+            //check the servlet hasn't been initialized before
+            int val = initCount.addAndGet(1);
+            assertThat(val, Matchers.equalTo(1));
+        }
+
+        @Override
+        public void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
+        {
+            if (initCount.get() != 1)
+            {
+                resp.sendError(500, "Servlet not initialized!");
+            }
+        }
+    }
+    
+    private static class AsyncResponseListener implements Response.CompleteListener
+    {
+        private CountDownLatch resultsLatch;
+        private Integer[] results;
+        private AtomicInteger index = new AtomicInteger();
+        
+        public AsyncResponseListener(int count)
+        {
+            resultsLatch = new CountDownLatch(count);
+            results = new Integer[count];
+        }
+        
+        public void onComplete(Result result)
+        {
+            results[index.getAndAdd(1)] = result.getResponse().getStatus();
+            resultsLatch.countDown();
+        }
+        
+        public void awaitCompletion() throws InterruptedException
+        {
+            assertTrue(resultsLatch.await(60L, TimeUnit.SECONDS));
+        }
+        
+        public Integer[] getResults()
+        {
+            return results;
+        }
+    }
+
+    @Test
+    public void testServletInitialization() throws Exception
+    {
+        Server server = new Server(0);
+        ServletContextHandler context = new ServletContextHandler(server, "/");
+        server.setHandler(context);
+        //add a lazily instantiated servlet
+        context.addServlet(new ServletHolder(DemoServlet.class), "/*");
+        server.start();
+        final int port = ((NetworkConnector)server.getConnectors()[0]).getLocalPort();
+        HttpClient client = new HttpClient();
+        try
+        {
+            client.start();
+
+            //Expect 2 responses
+            final AsyncResponseListener l = new AsyncResponseListener(2);
```

Comment:
Create the latch here and pass in the latch instance, so that you have it available here in the test method.
No need for `final`.